src/ranking.jl

```diff
@@ -171,3 +171,170 @@ Missing values are assigned rank `missing`.
 """
 tiedrank(x::AbstractArray; sortkwargs...) =
     _rank(_tiedrank!, x, Float64; sortkwargs...)
+
+"""
+    quantilerank(v, value; method=:inc)
+
+Compute the quantile(s)-position in [0-1] of a `value` relative to a collection `v`, e.g. a 
+quantile rank of x means that (x*100)% of the elements in `v` are lesser or lesser-equal the 
+given `value`. 
+
+The function gets the counts of elements of `v` that are less than `value` (`count_less`), 
+elements of `v` that are equal to `value` (`count_equal`) and the length of `v` (`n`). 
+It also extracts the highest value of elements below `value` (`last_less`) and the lowest 
+value of elements above `value` (`less_greater`). Using them, different methods and 
+definitions of the `quantilerank` are obtained by changing the `method` keyword argument: 
+
+`:inc` (default) - It calculates a value in the range 0 to 1 inclusive. 
+If `value ∈ v`, it returns `count_less / (n - 1)`, if not, apply interpolation based in 
+def. 7 in Hyndman and Fan (1996). 
+(equivalent to Excel `PERCENTRANK` and `PERCENTRANK.INC`)
+
+`:exc` - It calculates a value in the range 0 to 1 exclusive. 
+If `value ∈ v`, it returns `(count_less + 1) / (n + 1)`,  if not, apply interpolation 
+based in def. 6 in Hyndman and Fan (1996). 
+(equivalent to Excel `PERCENTRANK.EXC`)
+
+`:compete` - Based on the `competerank` of StatsBase, if `value ∈ v`, it returns 
+`count_less/ (n - 1)`, if not, returns `(count_less - 1) / (n - 1)`.
+Also, there is no interpolation.
+(equivalent to MariaDB `PERCENT_RANK`, dplyr `percent_rank`)
+
+`:tied` - Based in the def. in Roscoe, J. T. (1975), it returns 
+`(count_less + count_equal/2) / n` and there is no interpolation. 
+(equivalent to `mean` argument of Scipy `percentileofscore`)
+
+`:strict` - It returns `count_less / n` and there is no interpolation.
+(equivalent to `strict` method of Scipy `percentileofscore`)
+
+`:weak` - It returns `(count_less + count_equal) / n` and there is no interpolation.
+(equivalent to `weak` method of Scipy `percentileofscore`)
+
+!!! note
+    An `ArgumentError` is thrown if `v` contains `NaN` or `missing` values
+    and if `v` is empty or contains only one element.
+
+The function also supports types that have the mathematical properties of rings or total 
+order.
```

Comment:
```suggestion
```

I think now it can be removed as we added descriptions of the algorithms so it is clear when what properties of values are required.